torchvision/datasets/utils.py

```diff
@@ -78,6 +79,11 @@ def download_url(url, root, filename=None, md5=None):
     if check_integrity(fpath, md5):
         print('Using downloaded and verified file: ' + fpath)
     else:   # download the file
+        # set header
+        if header is not None:
+            opener = urllib.request.build_opener()
+            opener.addheaders = header
+            urllib.request.install_opener(opener)
```

Comment:
I wonder if we should revert back to the previous opener after the function has finished? Otherwise this globally affects the rest of the code using `urllib`. Not sure if it's possible to do it with the current API though.

Reply:
That makes sense and thanks for pointing this out!
It seems to work, if I create an empty `opener` and just install it again, as I get the same `forbidden` error then.